src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
 {
     public struct MemoryPoolIterator2
     {
+        private const int _maxHeaderStackLength = 16384;
```

Comment:
Its to prevent the stack blowing up; so if length is larger than this it heap allocates instead. Maybe a rename like `_maxStackAllocBytes`?